src/main/kotlin/eu/europa/ec/eudi/pidissuer/port/input/IssueCredential.kt

```diff
@@ -302,12 +301,10 @@ private val log = LoggerFactory.getLogger(IssueCredential::class.java)
  * Usecase for issuing a Credential.
  */
 class IssueCredential(
-    private val clock: Clock,
     private val credentialIssuerMetadata: CredentialIssuerMetaData,
     private val resolveCredentialRequestByCredentialIdentifier: ResolveCredentialRequestByCredentialIdentifier,
-    private val loadCNonceByAccessToken: LoadCNonceByAccessToken,
-    private val genCNonce: GenerateCNonce,
-    private val upsertCNonce: UpsertCNonce,
+    private val generateCNonce: GenerateCNonce,
```

Comment:
I was expecting a single `generateCNonce` that hides the how this is done.

`IssueCredential` doens't have to know how `c_nonce` is being generated

Reply:
Would something like this make sense?


```kotlin
fun interface GenerateCNonce {
  //Return the `serialized encrypted jwt` cnonce and `expires in`
  suspend operator fun invoke(): Pair<String, Duration>
}

fun interface VerifyCNonce {
  //Decrypts jwt, verifies cnonce is not expires, and returns the cnonce value
  suspend operator fun invoke(value: String): Result<Nonce>
}
```

`Nonce` as defined in Nimbus.

Reply:
If we follow this approach, we can remove the `CNonce` data class as well.